src/common/utility.cpp

```diff
@@ -277,15 +279,18 @@ bool Utility::fsCasePreserving()
 
 bool Utility::fileNamesEqual(const QString &fn1, const QString &fn2)
 {
-    const QDir fd1(fn1);
-    const QDir fd2(fn2);
-
+    const QFileInfo fd1(fn1);
+    const QFileInfo fd2(fn2);
+    qDebug() << fd1.canonicalFilePath() << fd2.canonicalFilePath();
     // Attention: If the path does not exist, canonicalPath returns ""
     // ONLY use this function with existing pathes.
```

Comment:
I think the comment should be reworded a bit, now that this is actually checked (which generally is a good idea; comments in the implementation aren't really read by users).